src/widgets/Chat/useChat.jsx

```diff
@@ -71,23 +71,24 @@ const useChat = () => {
 
     // TODO Restructure project context
     const project = {
-      context: {
-        project: {
-          id,
-          type: "chat",
-          name: "Chat Project",
-          description: "This project has been converted from chat",
-        },
-        api,
-        functions,
-        declarations,
+      specification: {
+        api: api,
+        functions: functions,
+        declarations: declarations,
+        types: [],
+      },
+      project: {
+        id,
+        type: "CHAT",
+        name: "Chat Project",
+        description: "This project has been converted from chat",
       },
     };
 
-    localStorage.setItem(`ide.projects.${id}`, JSON.stringify(project));
+    localStorage.setItem(`ide.context.${id}`, JSON.stringify(project));
     publish("CHAT_CONVERTED", chat);
-    navigate(`/${id}/api?mode=local`);
     navigate(0);
```

Comment:
Can we do it without `navigate(0)`? We had an issue before, but it should be working now

Reply:
solved using startTransition